Expose SharedVariableMap to Scout JS

mvilliger opened the pull request:
Synchronize selected variables from the SharedVariableMap of the
ClientSession to the browser.

362057

mvilliger committed:
Expose SharedVariableMap to Scout JS

Synchronize selected variables from the SharedVariableMap of the
ClientSession to the browser.

362057

mvilliger committed:
Support maxRowCount and limitedResult for Scout JS PageWithTable

- Introduce TableMaxResultsHelper which adds maxRowCount property of the
  Table as MaxRowCountContributionDo to a DO.
  In a PageWithTable the new function _withMaxRowCountContribution must
  be called e.g. in _loadTableData to make use of the helper and add the
  MaxRowCountContributionDo.
- Add limited result TableStatus to Table.
- Add MaxResultsHelper and ResultLimiter which automatically reads
  maxRowCount properties sent by the client together with a server-side
  limit.
- Introduce LimitedResultInfoContributionDo which adds the metadata info
  of the limiting operation of the ResultLimiter to a DO. This metadata
  can then be used to display to the client (e.g. by using the new
  TableStatus).
- Update hellojs archetype to use new features

362057

cguglielmo commented on eclipse-scout-core/src/session/SessionAdapter.ts:
```diff
@@ -0,0 +1,24 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+import {ModelAdapter} from './ModelAdapter';
```
Import from index

cguglielmo commented on eclipse-scout-core/src/table/Table.ts:
```diff
@@ -4901,6 +4901,38 @@ export class Table extends Widget implements TableModel {
+    const showingRowCountText = decimalFormat.format(numRows);
+    const estimatedRowCountText = decimalFormat.format(estimatedRowCount);
+    let message: string;
+    if (Device.get().supportsOnlyTouch()) {
```
I think we should change this to Device.get().type === Device.Type.MOBILE, also in Java. I think this is only relevant for small screens

cguglielmo commented on org.eclipse.scout.rt.api.data/src/main/java/org/eclipse/scout/rt/api/data/table/MaxResultsHelper.java:
```diff
@@ -0,0 +1,174 @@
+     * ({@link LimitedResultInfoContributionDo#isLimitedResult()})</li>
+     * <li>The maximum number of entries that would have been allowed
+     * ({@link LimitedResultInfoContributionDo#getMaxRowCount()})</li>
+     * <li>({@link LimitedResultInfoContributionDo#getEstimatedRowCount()}) is currently not set and will always be <=
```
Maybe we should add a member, so that this feature can be enabled? I tried to test it but I cannot enable it

mvilliger committed:
Add optional dependency to annotation processor

This ensures in a parallel full rebuild the APT module is build before
the module using it.
